examples/catboost_simple.py

```diff
@@ -0,0 +1,66 @@
+"""
+Optuna example that optimizes a classifier configuration for cancer dataset using
+Catboost.
+
+In this example, we optimize the validation accuracy of cancer detection using
+Catboost. We optimize both the choice of booster model and their hyperparameters.
+
+We have following two ways to execute this example:
+
+(1) Execute this code directly.
+    $ python catboost_simple.py
+
+
+(2) Execute through CLI.
+    $ STUDY_NAME=`optuna create-study --direction maximize --storage sqlite:///example.db`
+    $ optuna study optimize catboost_simple.py objective --n-trials=100 --study $STUDY_NAME \
+      --storage sqlite:///example.db
+
+"""
+
+import catboost as cb
+import numpy as np
+import sklearn.datasets
+import sklearn.metrics
+from sklearn.model_selection import train_test_split
+
+import optuna
+
+
+def objective(trial):
+    data, target = sklearn.datasets.load_breast_cancer(return_X_y=True)
+    train_x, test_x, train_y, test_y = train_test_split(data, target, test_size=0.3)
+
+    param = {
+        'objective': trial.suggest_categorical('objective', ['Logloss', 'CrossEntropy']),
+        'iterations': trial.suggest_int('num_leaves', 500, 2000),
+        'colsample_bylevel': trial.suggest_uniform('colsample_bylevel', 0.01, 0.1),
+        'depth': trial.suggest_int('depth', 1, 16),
+        'boosting_type': trial.suggest_categorical('boosting_type', ['Ordered', 'Plain']),
+        'learning_rate': trial.suggest_loguniform('learning_rate', 1e-6, 1.0)
+    }
+
```

Comment:
> Would you like me to add more parameters to tune from the Catboost docs, or do you think this is sufficient ?

IMO, it's better to add suggestions of conditional parameters to show the define-by-run feature of Optuna.
For example,
```python
bootstrap_type = trial.suggest_categorical('bootstrap_type', ['Bayesian', 'Bernoulli', 'MVS'])
params['bootstrap_type'] = bootstrap_type

if bootstrap_type == 'Bayesian':
    params['bagging_temperature'] = trial.suggest_uniform('bagging_temperature', 0, 10)
elif bootstrap_type == 'Bernoulli':
   params['subsample'] = trial.suggest_uniform('subsample', 0, 1)
elif bootstrap_type == 'MVS':
   params['mvs_head_fraction'] = trial.suggest_uniform('mvs_head_fraction', 0, 1)
```

**[NOTE]** I'm unfamiliar with Catboost, so I don't know the details of the above parameters.

Reply:
Thanks for the feedback @sile, the `mvs_head_fraction` is giving me some issues although I am using the most recent version of catboost (v0.16.4). Should I remove it from the example ?

Reply:
Sure ! Just removed it